pkg/csi/manager.go

```diff
@@ -81,3 +116,43 @@ func (m *Manager) Run(cfg *config.Config) error {
 
 	return nil
 }
+
+func discoverVMIName(nodeID string, vmis kubecli.VirtualMachineInstanceInterface, ifaces []sysfsnet.Interface) (string, error) {
+	if len(ifaces) == 0 {
+		return "", errVMINotFound
+	}
+
+	macs := make(map[string]struct{})
```

Comment:
After golang 1.18, the slice support `Contains`.
Maybe we could avoid using map?
Reference: https://pkg.go.dev/golang.org/x/exp/slices#Contains

WDYT?

Reply:
> Maybe we could avoid using map?

Is there a reason for doing so? I used map for efficient lookups.

Reply:
If we use the map for lookups, we need to create a map, which will increase overhead dynamically.
If we reuse the slice, we could reduce the above overhead.

You could refer to the following benchmark:
```
package main

import (
	"testing"

	"golang.org/x/exp/slices"
)

var checkedMacSlice = []string{
	"ab:ab:ab:ab:ab:ab",
	"ce:ce:ce:ce:ce:ce",
}

var checkedMacSlice2 = []string{
	"11:11:11:11:11:11",
	"22:22:22:22:22:22",
}

var targetSlice = []string{
	"ab:ab:ab:ab:ab:ab",
	"ce:ce:ce:ce:ce:ce",
}

func lookupMapFound() bool {
	targetMap := make(map[string]struct{})
	for _, iface := range targetSlice {
		targetMap[iface] = struct{}{}
	}
	for _, mac := range checkedMacSlice {
		if _, ok := targetMap[mac]; ok {
			return true
		}
	}
	return false
}

func lookupSliceFound() bool {
	for _, mac := range checkedMacSlice {
		if slices.Contains(targetSlice, mac) {
			return true
		}
	}
	return false
}

func lookupMapNotFound() bool {
	targetMap := make(map[string]struct{})
	for _, iface := range targetSlice {
		targetMap[iface] = struct{}{}
	}
	for _, mac := range checkedMacSlice2 {
		if _, ok := targetMap[mac]; ok {
			return true
		}
	}
	return false
}

func lookupSliceNotFound() bool {
	for _, mac := range checkedMacSlice2 {
		if slices.Contains(targetSlice, mac) {
			return true
		}
	}
	return false
}

func BenchmarkLookupMapFound(b *testing.B) {
	for i := 0; i < b.N; i++ {
		lookupMapFound()
	}
}

func BenchmarkLookupSliceFound(b *testing.B) {
	for i := 0; i < b.N; i++ {
		lookupSliceFound()
	}
}

func BenchmarkLookupMapNotFound(b *testing.B) {
	for i := 0; i < b.N; i++ {
		lookupMapNotFound()
	}
}

func BenchmarkLookupSliceNotFound(b *testing.B) {
	for i := 0; i < b.N; i++ {
		lookupSliceNotFound()
	}
}
```
Result:
```
$ go test -v -bench=.
goos: linux
goarch: amd64
pkg: viperSample/lookup-bench
cpu: 12th Gen Intel(R) Core(TM) i7-12700K
BenchmarkLookupMapFound
BenchmarkLookupMapFound-20              59903853                20.22 ns/op
BenchmarkLookupSliceFound
BenchmarkLookupSliceFound-20            487960608                2.477 ns/op
BenchmarkLookupMapNotFound
BenchmarkLookupMapNotFound-20           34367911                29.36 ns/op
BenchmarkLookupSliceNotFound
BenchmarkLookupSliceNotFound-20         100000000               12.09 ns/op
PASS
```